stdlib/SparseArrays/src/sparsematrix.jl

```diff
@@ -227,22 +227,36 @@ end
 Base.replace_in_print_matrix(A::AbstractSparseMatrix, i::Integer, j::Integer, s::AbstractString) =
     Base.isstored(A, i, j) ? s : Base.replace_with_centered_mark(s)
 
-function Base.show(io::IO, ::MIME"text/plain", S::AbstractSparseMatrixCSC)
+function Base.array_summary(io::IO, S::AbstractSparseMatrixCSC, dims::Tuple{Vararg{Base.OneTo}})
     _checkbuffers(S)
     xnnz = nnz(S)
     m, n = size(S)
     print(io, m, "×", n, " ", typeof(S), " with ", xnnz, " stored ",
               xnnz == 1 ? "entry" : "entries")
-    if !(m == 0 || n == 0)
-        print(io, ":")
-        show(IOContext(io, :typeinfo => eltype(S)), S)
+    nothing
+end
+
+# called by `show(io, MIME("text/plain"), ::AbstractSparseMatrixCSC)`
+function Base.print_array(io::IO, S::AbstractSparseMatrixCSC)
+    if max(size(S)...) < 16
+        Base.print_matrix(io, S)
+    else
+        _show_with_braille_patterns(io, S)
     end
 end
 
-Base.show(io::IO, S::AbstractSparseMatrixCSC) = Base.show(convert(IOContext, io), S::AbstractSparseMatrixCSC)
+# called by `show(io, ::AbstractSparseMatrixCSC)`
+function Base._show_nonempty(io::IO, S::AbstractSparseMatrixCSC, prefix::String)
+    _checkbuffers(S)
+    if max(size(S)...) < 16
+        Base._show_nonempty(io, Base.inferencebarrier(S), prefix, false, axes(S))
+    else
+        _show_with_braille_patterns(io, S)
```

Comment:
Good, question, I wasn't really sure about that either. OTOH if we don't use Braille, the `show` output will likely just be a bunch of zeros for any sufficiently large sparse matrix.

Reply:
I don't care too much about the print/repr output except that we avoid that all the elements end up being printed for large arrays e.g. when the matrix is a field in a struct or in error messages. It can make you process hang forever and/or produce extremely large output.

Reply:
OK, in that case I can just delete this special case and make it always use the default `AbstractArray` printing.

Reply:
Perhaps the `repr` output should be of the form `sparse(I,J,K)`

Reply:
How about we still use the generic AbstractArray printing for sizes `<= 4x4`, but use one using `sparse` for anything greater than that?

Reply:
I feel like the `repr` output should normally be something terse that reproduces the input if possible.  If people have a small sparse matrix and they want to print the dense version they can always call `Matrix(...)` on it.